src/ffi/mod.rs

```diff
@@ -401,6 +415,15 @@ pub extern "C" fn telio_get_adapter_luid(dev: &telio) -> u64 {
     }
 }
 
+fn char_ptr_to_type<T: std::str::FromStr>(private_key: *const c_char) -> Result<T, telio_result> {
+    let cstr = unsafe { CStr::from_ptr(private_key) };
+    let private_key = cstr.to_str().map_err(|_| TELIO_RES_INVALID_STRING);
```

Comment:
nit: Adding `?` at the end of this line will make the code simpler since you will get the `key` directly.